### ISSUE TITLE
`undocumented_unsafe_blocks` can be broken by `rustfmt`

### ISSUE THREAD

**Issue by clarfonthey** [rust-lang/rust-clippy#13024]
### Summary

Consider the following code:

```rust
// SAFETY: trust me ok
long_var_name = unsafe {
    /* ... */
};
```

Normally, `rustfmt` will either leave the `unsafe` block as above or convert this into the following, depending on its length:

```rust
// SAFETY: trust me ok
long_var_name = unsafe { /* ... */ };
```

However, in rare cases, if `/* ... */` is exactly the right length, `rustfmt` will emit:

```rust
// SAFETY: trust me ok
long_var_name =
    unsafe { /* ... */ };
```

Which breaks the `undocumented_unsafe_blocks` lint, since the comment isn't immediately before the line with `unsafe`. It should accept cases like this.

### Lint Name

undocumented_unsafe_blocks

**Comment by notriddle** [comment]
When [I run this](<https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2810b2c76d8bf47978d785476af7d023>) with a current version of clippy, the third test doesn't lint thanks to the `accept-comment-above-statement` option, [which is turned on by default](<https://github.com/rust-lang/rust-clippy/pull/11170>).

This feature was actually merged before this issue was opened, and made its way to stable in [November 2023](<https://github.com/rust-lang/rust-clippy/blob/7cda242e3ce3eaf59223f3921e7eb53b15c9e2a5/CHANGELOG.md#enhancements-10>).
